lib/pos_random_seed.go

```diff
@@ -93,20 +103,30 @@ func (bav *UtxoView) VerifyRandomSeedSignature(
 	if err != nil {
 		return nil, errors.Wrapf(err, "UtxoView.VerifyRandomSeedSignature: problem retrieving CurrentRandomSeedHash: ")
 	}
-	isVerified, err := signerPublicKey.Verify(randomSeedSignature, currentRandomSeedHash[:])
+	randomSeedHash, err := verifyRandomSeedSignatureGivenRandomSeedHash(signerPublicKey, randomSeedSignature, currentRandomSeedHash)
 	if err != nil {
 		return nil, errors.Wrapf(err, "UtxoView.VerifyRandomSeedSignature: problem verifying RandomSeedSignature: ")
 	}
+	return randomSeedHash, nil
+}
+
+func verifyRandomSeedSignatureGivenRandomSeedHash(
+	signerPublicKey *bls.PublicKey, randomSeedSignature *bls.Signature, randomSeedHash *RandomSeedHash,
+) (*RandomSeedHash, error) {
+	isVerified, err := signerPublicKey.Verify(randomSeedSignature, randomSeedHash[:])
```

Comment:
This function just performs a signature verification. Rather than naming it based on its purpose, it's clearer to instead describe what the function does.

I'm also not sure why MF had this verification function return the next random seed hash. It's clearer to separate this function into two separate ones: one that performs the signature verification, and another that computes the next random seed hash

```suggestion
func verifySignatureOnRandomSeedHash(
	signerPublicKey *bls.PublicKey, randomSeedSignature *bls.Signature, randomSeedHash *RandomSeedHash,
) (bool, error) {
	return signerPublicKey.Verify(randomSeedSignature, randomSeedHash[:])
}
```